Assets/Scripts/Persona/api/APIClient.cs

```diff
@@ -113,5 +111,62 @@ public async Task<string> MakePostRequestAsync(string url, string postData, Dict
                 }
             }
         }
+
+        public async Task<Texture2D> DownloadImageFromUrl(string mediaUrl)
+        {
+            SentrySdk.AddBreadcrumb(message: "Beginning DownloadImageFromUrl", category: "sdk.milestone", level: BreadcrumbLevel.Info);
+            using (UnityWebRequest request = UnityWebRequestTexture.GetTexture(mediaUrl))
+            {
+                SentrySdk.AddBreadcrumb(message: $"DownloadImageFromUrl url - {mediaUrl}", category: "sdk.milestone", level: BreadcrumbLevel.Info);
+                var asyncOperation = request.SendWebRequest();
+                while (!asyncOperation.isDone)
+                {
+                    await Task.Yield(); // Yield to prevent blocking the main thread
+                }
+
+                if (request.result == UnityWebRequest.Result.ConnectionError || request.result == UnityWebRequest.Result.ProtocolError)
+                {
+                    throw new Exception(request.error);
+                    //Debug.LogError("Image download error: " + request.error);
+                    //return null;
+                }
+                else
+                {
+                    Texture2D downloadedTexture = DownloadHandlerTexture.GetContent(request);
+                    return downloadedTexture;
+                }
+            }
+        }
+
+        public async Task<byte[]> DownloadGifFromUrl(string url)
```

Comment:
This function too looks like a utility function

Reply:
Yes, same reason stated above